libraries/AP_BoardConfig/px4_drivers.cpp

```diff
@@ -196,17 +213,297 @@ void AP_BoardConfig::px4_setup_canbus(void)
+        } else {
+            if (px4_start_driver(mpu9250_main, "mpu9250", "-R 14 start")) {
+                printf("Found MPU9250 internal\n");
+            } else {
+                px4_sensor_error("No MPU6000 or MPU9250");
+            }
+        }
+        if (px4_start_driver(hmc5883_main, "hmc5883", "-C -T -S -R 8 start")) {
+            printf("Found SPI hmc5883\n");
+        }
+    } else {
+        // not FMUV3 (ie. not a pixhawk2)
+        if (px4_start_driver(mpu6000_main, "mpu6000", "start")) {
+            printf("Found MPU6000\n");
+        } else {
+            if (px4_start_driver(mpu9250_main, "mpu9250", "start")) {
+                printf("Found MPU9250\n");
+            } else {
+                printf("No MPU6000 or MPU9250\n");
+            }
+        }
+        if (px4_start_driver(l3gd20_main, "l3gd20", "start")) {
+            printf("l3gd20 started OK\n");
+        } else {
+            px4_sensor_error("no l3gd20 found");
+        }
+        if (px4_start_driver(lsm303d_main, "lsm303d", "-a 16 start")) {
+            printf("lsm303d started OK\n");
+        } else {
+            px4_sensor_error("no lsm303d found");
+        }
+    }
+    printf("FMUv2 sensors started\n");
+#endif // CONFIG_ARCH_BOARD_PX4FMU_V2
+}
+
+/*
+  setup sensors for PX4v1
+ */
+void AP_BoardConfig::px4_start_fmuv1_sensors(void)
+{
+#if defined(CONFIG_ARCH_BOARD_PX4FMU_V1)
+    printf("Starting FMUv1 sensors\n");
+    if (px4_start_driver(hmc5883_main, "hmc5883", "-C -T -X start")) {
+            printf("Have external hmc5883\n");
+    } else {
+        printf("No external hmc5883\n");
+    }
+    if (px4_start_driver(hmc5883_main, "hmc5883", "-C -T -I  start")) {
+        printf("Have internal hmc5883\n");
+    } else {
+        printf("No internal hmc5883\n");
+    }
+    if (px4_start_driver(mpu6000_main, "mpu6000", "start")) {
+        printf("mpu6000 started OK\n");
+    } else {
+        px4_sensor_error("mpu6000");
+    }
+    if (px4_start_driver(l3gd20_main, "l3gd20", "start")) {
+        printf("l3gd20 started OK\n");
+    } else {
+        printf("No l3gd20\n");
+    }
+#endif // CONFIG_ARCH_BOARD_PX4FMU_V1
+}
+
+/*
+  setup sensors for FMUv4
+ */
+void AP_BoardConfig::px4_start_fmuv4_sensors(void)
+{
+#if defined(CONFIG_ARCH_BOARD_PX4FMU_V4)
+    printf("Starting FMUv4 sensors\n");
+    if (px4_start_driver(hmc5883_main, "hmc5883", "-C -T -X start")) {
+        printf("Have external hmc5883\n");
+    } else {
+        printf("No external hmc5883\n");
+    }
+    if (px4_start_driver(hmc5883_main, "hmc5883", "-C -T -S -R 2 start")) {
+        printf("Have SPI hmc5883\n");
+    } else {
+        printf("No SPI hmc5883\n");
+    }
+
+    if (px4_start_driver(mpu6000_main, "mpu6000", "-R 2 -T 20608 start")) {
+        printf("Found ICM-20608 internal\n");
+    }
+
+    if (px4_start_driver(mpu9250_main, "mpu9250", "-R 2 start")) {
+        printf("Found mpu9250 internal\n");
+    }
+#endif // CONFIG_ARCH_BOARD_PX4FMU_V4
```

Comment:
again it is because that is how rc.APM behaved. I'm trying for a "no surprises" change to the new structure, with later updates to start adding appropriate error checks.